core/storage/user/gae_models.py

```diff
@@ -148,6 +148,20 @@ class UserStatsModel(base_models.BaseMapReduceBatchResultsModel):
     # The impact score.
     impact_score = ndb.FloatProperty(indexed=True)
 
+class UserDashboardStatsModel(base_models.BaseMapReduceBatchResultsModel):
```

Comment:
Actually, the existing impact aggregator job makes use of ratings and plays in its calculation. It seems to me like you should be able to reuse it, and get it to calculate more stuff?